core/trino-main/src/main/java/io/trino/server/ui/ClusterResource.java

```diff
@@ -36,19 +36,21 @@ public class ClusterResource
     private final NodeVersion version;
     private final String environment;
     private final long startTime = System.nanoTime();
+    private final WebUiConfig webUiConfig;
 
     @Inject
-    public ClusterResource(NodeVersion nodeVersion, NodeInfo nodeInfo)
+    public ClusterResource(NodeVersion nodeVersion, NodeInfo nodeInfo, WebUiConfig webUiConfig)
     {
         this.version = requireNonNull(nodeVersion, "nodeVersion is null");
         this.environment = nodeInfo.getEnvironment();
+        this.webUiConfig = requireNonNull(webUiConfig, "webUiConfig is null");
```

Comment:
instead of assigning config itself, just assign value of `isPreviewEnabled`
so:

```
this.previewUiEnabled = webUiConfig.isPreviewEnabled();
```

rnn check is not required as it will throw NPE immediately in the constructor due to a chained call.